python/bin/repl_stub.py

```diff
@@ -1,3 +1,18 @@
+"""Simulates the REPL that Python spawns when invoking the binary with no arguments.
+
+The code module is responsible for the default shell.
+
+The import and `ocde.interact()` call here his is equivalent to doing:
+
+    $ python3 -m code
+    Python 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0] on linux
+    Type "help", "copyright", "credits" or "license" for more information.
+    (InteractiveConsole)
+    >>>
+
+The logic for PYTHONSTARTUP is handled in python/private/repl_template.py.
+"""
+
```

Comment:
Could we also copy the logic for the banner, so that users are greeted with the familiar Python version?

Reply:
Done.

That is actually surprisingly awkward. That code is implemented a few times. At least once in the cpython startup code and once in the `code` module.  Since there's no good way to invoke either of those from here, I did indeed end up copying it. Let me know if this is what you had in mind.

Reply:
Sorry, did not realize it was a little awkward, but thanks for doing it!